mantidimaging/core/operations/clip_values/clip_values.py

```diff
@@ -55,27 +57,30 @@ def filter_func(data,
         # We're using is None because 0.0 is a valid value
         if clip_min is None and clip_max is None:
             raise ValueError('At least one of clip_min or clip_max must be supplied')
+        params = {
+            'clip_min': clip_min,
+            'clip_max': clip_max,
+            'clip_min_new_value': clip_min_new_value,
+            'clip_max_new_value': clip_max_new_value
+        }
 
-        progress = Progress.ensure_instance(progress, num_steps=2, task_name='Clipping Values.')
-        with progress:
-            sample = data.data
-            progress.update(msg="Determining clip min and clip max")
-            clip_min = clip_min if clip_min is not None else sample.min()
-            clip_max = clip_max if clip_max is not None else sample.max()
-
-            clip_min_new_value = clip_min_new_value if clip_min_new_value is not None else clip_min
-
-            clip_max_new_value = clip_max_new_value if clip_max_new_value is not None else clip_max
-
-            progress.update(msg=f"Clipping data with values min {clip_min} and max {clip_max}")
-
-            # this is the fastest way to clip the values, np.clip does not do
-            # the clipping in place and ends up copying the data
-            sample[sample < clip_min] = clip_min_new_value
-            sample[sample > clip_max] = clip_max_new_value
+        ps.run_compute_func(ClipValuesFilter.compute_function, data.data.shape[0], [data.shared_array], params,
+                            progress)
 
         return data
 
+    @staticmethod
+    def compute_function(i: int, arrays: List[np.ndarray], params: Dict[str, Any]):
```

Comment:
The type of `arrays` here should be `np.ndarray`. This is because if a single array is passed to `run_compute` it get unpacked from the list. So maybe do it as
```
def compute_function(i: int, array: np.ndarray, params: Dict[str, Any]):
    slice = array[i]
```